pennylane/transforms/compile.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Code for the high-level quantum function transform that executes compilation."""
+
+from inspect import getmembers, isclass
+
+from pennylane import apply
+from pennylane.tape import get_active_tape
+
+from pennylane.transforms import qfunc_transform
+from pennylane.transforms.optimization import cancel_inverses, commute_controlled, merge_rotations
+
+from pennylane import templates
+
+# Get a list of the names of existing templates in PennyLane
+template_list = [cls[0] for cls in getmembers(templates, isclass)]
+
+
+default_pipeline = [commute_controlled, cancel_inverses, merge_rotations]
+
+
+@qfunc_transform
+def compile(tape, pipeline=None, basis_set=None, num_passes=1):
+    """Compile a circuit by applying a series of transforms to a quantum function.
+
+    The default set of transforms includes (in order):
+     - pushing all commuting single-qubit gates as far right as possible
+       (:func:`~.pennylane.transforms.optimization.commute_controlled`)
+     - cancellation of adjacent inverse gates
+       (:func:`~.pennylane.transforms.optimization.cancel_inverses`)
+     - merging adjacent rotations of the same type
+       (:func:`~.pennylane.transforms.optimization.merge_rotations`)
```

Comment:
```suggestion
    The default set of transforms includes (in order):

    - pushing all commuting single-qubit gates as far right as possible
      (:func:`~.commute_controlled`)
    - cancellation of adjacent inverse gates
      (:func:`~.cancel_inverses`)
    - merging adjacent rotations of the same type
      (:func:`~.merge_rotations`)
```

Two changes:

- Unindented the bullet list (an indentation block is a glossary item in ReST)
- Changed the function references, since the links aren't working for me in the rendered docs (and hoping this fixes it)